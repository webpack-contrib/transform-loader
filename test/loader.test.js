/* eslint-disable
  prefer-destructuring,
*/
import loader from '../src/cjs';
import webpack from './helpers/compiler';

describe('Loader', () => {
  test('Defaults', async () => {
    const config = {
      loader: {
        test: /\.coffee$/,
        loader: `${loader}?coffeeify`,
        options: {},
      },
    };

    const stats = await webpack('fixture.js', config);
    const { source } = stats.toJson().modules[1];

    expect(source).toMatchSnapshot();
  });
});
