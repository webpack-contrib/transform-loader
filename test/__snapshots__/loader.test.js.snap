// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Defaults 1`] = `"throw new Error(\\"Module build failed: TypeError: Cannot read property 'indexOf' of undefined\\\\n    at Resolver.parse (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/Resolver.js:181:28)\\\\n    at Resolver.<anonymous> (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/ParsePlugin.js:14:25)\\\\n    at Resolver.applyPluginsAsyncSeriesBailResult1 (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/tapable/lib/Tapable.js:256:13)\\\\n    at runNormal (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/Resolver.js:130:20)\\\\n    at Resolver.doResolve (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/Resolver.js:116:3)\\\\n    at Resolver.<anonymous> (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:37:12)\\\\n    at Resolver.applyPluginsAsyncSeriesBailResult1 (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/tapable/lib/Tapable.js:256:13)\\\\n    at runNormal (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/Resolver.js:130:20)\\\\n    at Resolver.doResolve (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/Resolver.js:116:3)\\\\n    at Resolver.resolve (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/enhanced-resolve/lib/Resolver.js:86:14)\\\\n    at Object.resolve (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/node_modules/webpack/lib/NormalModule.js:133:14)\\\\n    at Object.<anonymous>.exports.default (/Users/d3viant0ne/projects/oss/webpack-contrib/transform-loader/src/index.js:25:10)\\");"`;
